tests/flytekit/unit/core/test_python_function_task.py

```diff
@@ -122,3 +125,83 @@ def foo_missing_cache_version(i: str):
         @task(cache_serialize=True)
         def foo_missing_cache(i: str):
             print(f"{i}")
+
+
+def test_pod_template():
```

Comment:
Can you also add a test where no pod spec is defined?

Reply:
I changed `task_with_minimum_pod_template` to test the case without podspec. The original intention of `task_with_minimum_pod_template` to test pod spec's volume is already covered in `test_pod_template`.